schemaVersion: 2.0.0

fileExistenceTests:
  - name: 'ptvsd for python 2.7'
    path: '/duct-tape/python/lib/python2.7/site-packages/ptvsd/__init__.py'
  - name: 'debugpy for python 2.7'
    path: '/duct-tape/python/lib/python2.7/site-packages/debugpy/__init__.py'
  - name: 'pydevd for python 2.7'
    path: '/duct-tape/python/pydevd/python2.7/lib/python2.7/site-packages/pydevd.py'
  - name: 'pydevd-pycharm for python 2.7'
    path: '/duct-tape/python/pydevd-pycharm/python2.7/lib/python2.7/site-packages/pydevd.py'

  - name: 'ptvsd for python 3.5'
    path: '/duct-tape/python/lib/python3.5/site-packages/ptvsd/__init__.py'
  - name: 'debugpy for python 3.5'
    path: '/duct-tape/python/lib/python3.5/site-packages/debugpy/__init__.py'
  - name: 'pydevd for python 3.5'
    path: '/duct-tape/python/pydevd/python3.5/lib/python3.5/site-packages/pydevd.py'
  - name: 'pydevd-pycharm for python 3.5'
    path: '/duct-tape/python/pydevd-pycharm/python3.5/lib/python3.5/site-packages/pydevd.py'

  - name: 'ptvsd for python 3.6'
    path: '/duct-tape/python/lib/python3.6/site-packages/ptvsd/__init__.py'
  - name: 'debugpy for python 3.6'
    path: '/duct-tape/python/lib/python3.6/site-packages/debugpy/__init__.py'
  - name: 'pydevd for python 3.6'
    path: '/duct-tape/python/pydevd/python3.6/lib/python3.6/site-packages/pydevd.py'
  - name: 'pydevd-pycharm for python 3.6'
    path: '/duct-tape/python/pydevd-pycharm/python3.6/lib/python3.6/site-packages/pydevd.py'

  - name: 'ptvsd for python 3.7'
    path: '/duct-tape/python/lib/python3.7/site-packages/ptvsd/__init__.py'
  - name: 'debugpy for python 3.7'
    path: '/duct-tape/python/lib/python3.7/site-packages/debugpy/__init__.py'
  - name: 'pydevd for python 3.7'
    path: '/duct-tape/python/pydevd/python3.7/lib/python3.7/site-packages/pydevd.py'
  - name: 'pydevd-pycharm for python 3.7'
    path: '/duct-tape/python/pydevd-pycharm/python3.7/lib/python3.7/site-packages/pydevd.py'

  - name: 'ptvsd for python 3.8'
    path: '/duct-tape/python/lib/python3.8/site-packages/ptvsd/__init__.py'
  - name: 'debugpy for python 3.8'
    path: '/duct-tape/python/lib/python3.8/site-packages/debugpy/__init__.py'
  - name: 'pydevd for python 3.8'
    path: '/duct-tape/python/pydevd/python3.8/lib/python3.8/site-packages/pydevd.py'
  - name: 'pydevd-pycharm for python 3.8'
    path: '/duct-tape/python/pydevd-pycharm/python3.8/lib/python3.8/site-packages/pydevd.py'

  - name: 'ptvsd for python 3.9'
    path: '/duct-tape/python/lib/python3.9/site-packages/ptvsd/__init__.py'
  - name: 'debugpy for python 3.9'
    path: '/duct-tape/python/lib/python3.9/site-packages/debugpy/__init__.py'
  - name: 'pydevd for python 3.9'
    path: '/duct-tape/python/pydevd/python3.9/lib/python3.9/site-packages/pydevd.py'
  - name: 'pydevd-pycharm for python 3.9'
    path: '/duct-tape/python/pydevd-pycharm/python3.9/lib/python3.9/site-packages/pydevd.py'

  - name: 'python launcher'
    path: '/duct-tape/python/launcher'
    isExecutableBy: any

commandTests:
  - name: "run with no /dbg should fail"
    command: "sh"
    args: ["/install.sh"]
    expectedError: ["Error: installation requires a volume mount at /dbg"]
    exitCode: 1
  - name: "run with /dbg should install"
    setup: [["mkdir", "/dbg"]]
    command: "sh"
    args: ["/install.sh"]
    expectedOutput: ["Installing runtime debugging support files in /dbg", "Installation complete"]
    exitCode: 0
